src/plugins/unified_field_list/public/services/field_stats.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { lastValueFrom } from 'rxjs';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import type { DataPublicPluginStart } from '@kbn/data-plugin/public';
+import type { DataViewFieldBase } from '@kbn/es-query';
+import type { FieldStatsResponse } from '../../common/types';
+import {
+  fetchAndCalculateFieldStats,
+  SearchHandler,
+  buildSearchParams,
+} from '../../common/utils/field_stats_utils';
+
+interface FetchFieldStatsParams {
+  services: {
+    data: DataPublicPluginStart;
+  };
+  dataView: DataView;
+  field: DataViewFieldBase;
+  fromDate: string;
+  toDate: string;
+  dslQuery: object;
+  size?: number;
+  abortController?: AbortController;
+}
+
+/**
+ * Loads and aggregates stats data for a data view field
+ * @param services
+ * @param dataView
+ * @param field
+ * @param fromDate
+ * @param toDate
+ * @param dslQuery
+ * @param size
+ * @param abortController
+ */
+export const loadFieldStats = async ({
+  services,
+  dataView,
+  field,
+  fromDate,
+  toDate,
+  dslQuery,
+  size,
+  abortController,
+}: FetchFieldStatsParams): Promise<FieldStatsResponse<string | number>> => {
+  const { data } = services;
```

Comment:
Like with the react components - lets move this code and its imports into a separate module and lazily load it in here to reduce the page load bundle size.

This will probably become irrelevant once this is a package but for plugins everything exported on the top level `index.ts` file makes it into the main bundle even if it's only imported in async bundles elsewhere.